editoast/src/views/search/queryeval.rs

```diff
@@ -0,0 +1,889 @@
+                .map(QValue::Integer)
+                .map_err(|_| format!("could not convert {n} to i64").into()),
+            QueryExpr::SignedInt(n) => Ok(QValue::Integer(*n)),
+            QueryExpr::Float(n) => Ok(QValue::Float(*n)),
+            QueryExpr::String(s) => {
+                Ok(QValue::String(self.interpolation_placeholder(s).to_owned()))
+            }
+            QueryExpr::Column(name, table) => Ok(QValue::Column {
+                name: name.to_owned(),
+                table: table.clone().or(self.search_table_name.clone()),
+                spec: self
+                    .columns_type
+                    .get(name)
+                    .unwrap_or(&QType::Any.into())
+                    .to_owned(),
+            }),
+            QueryExpr::Call(function_name, query_args) => {
+                let args = query_args
+                    .iter()
+                    .map(|sub| self.eval(sub))
+                    .collect::<Result<Vec<QValue>, EvalError>>()?;
+                self.call(function_name, args)
+            }
+        }
+    }
+
+    pub fn call<S: Into<String>>(&self, function_name: S, args: Vec<QValue>) -> EvalResult<QValue> {
+        let function_name: String = function_name.into();
+        let values_types = args
+            .iter()
+            .map(|val| val.type_spec())
+            .collect::<Vec<TypeSpec>>();
+        let functions = self
+            .functions
+            .get(&function_name)
+            .ok_or(format!("undefined function '{function_name}'"))?;
+        let function = match functions.len() {
+            0 => Err(EvalError(format!("undefined function '{function_name}'"))),
+            1 => {
+                let function = functions.first().unwrap();
+                function
+                    .signature
+                    .typecheck_args(&values_types[..])
+                    .map_err(|TypeCheckError(err)| {
+                        EvalError(format!("in function '{0}': {err}", function.name))
+                    })?;
+                Ok(function)
+            }
+            _ => 'find_overload: {
+                for function in functions {
+                    if function.signature.typecheck_args(&values_types[..]).is_ok() {
+                        break 'find_overload Ok(function);
+                    }
+                }
+                let pprint = format!(
+                    "({0})",
+                    values_types
+                        .iter()
+                        .map(|ts| format!("{ts}"))
+                        .collect::<Vec<_>>()
+                        .join(", ")
+                );
+                Err(format!("no suitable overload of '{function_name}' found for {pprint}").into())
+            }
+        }?;
+        (function.fun)(args)
+            .map_err(|EvalError(msg)| format!("in function '{0}': {msg}", function.name).into())
+    }
+
+    /// Adds a function named `name` with the given `arity` into the environment.
```

Comment:
```suggestion
    /// Adds a function named `name` with the given `arity` into the environment
```
Main doc phrase should not end with a point according to the doc https://doc.rust-lang.org/beta/rust-by-example/meta/doc.html (not a big thing but I think we are mostly following this in editoast for now)